inheritance.rb

```diff
@@ -0,0 +1,42 @@
+class Person
+  @@hash  ={"Employee"  => "20K", "Senior_person" =>  "40K", "Customer" => "None"}
+
+  def initialize(name,post)
+    @name=name
+    @post=post
+  end
+
+  protected 
+  def protected_method
+    puts "salary of #{@name} is #{@@hash[@post]}"
+  end
+end
+
+class Employee < Person
+  def salary_eml
+    protected_method
+  end
+end
+
+class Senior_person < Person
+  def salary_snrp
+    protected_method
+  end
+end
+
+class Customer < Person
+  def salary_cstp
+    protected_method
+  end
+end
+
+
+emply = Employee.new("Niharika","Employee")
```

Comment:
here use space while passing arguments to method.
emply = Employee.new("Niharika", "Employee")